rs/ic-management-backend/src/subnets.rs

```diff
@@ -2,7 +2,44 @@ use std::collections::BTreeMap;
 
 use decentralization::{network::SubnetChange, SubnetChangeResponse};
 use ic_base_types::PrincipalId;
-use ic_management_types::{Node, TopologyChangeProposal};
+use ic_management_types::{Node, Status, Subnet, TopologyChangeProposal};
+use log::{info, warn};
+
+pub async fn unhealthy_with_nodes(
+    subnets: &BTreeMap<PrincipalId, Subnet>,
+    nodes_health: BTreeMap<PrincipalId, Status>,
+) -> BTreeMap<PrincipalId, Vec<Node>> {
+    subnets
+        .clone()
+        .into_iter()
+        .filter_map(|(id, subnet)| {
+            let unhealthy = subnet
+                .nodes
+                .into_iter()
+                .filter_map(|n| match nodes_health.get(&n.principal) {
+                    Some(health) => {
+                        if *health == ic_management_types::Status::Healthy {
+                            None
+                        } else {
+                            info!("Node {} is {:?}", n.principal, health);
+                            Some(n)
+                        }
+                    }
+                    None => {
+                        warn!("Node {} has no known health, assuming unhealthy", n.principal);
+                        Some(n)
+                    }
```

Comment:
AFAIK we run backend as a child process and we can't really see the logs coming from it so the `info!` and `warn!` aren't really visible right? If they are not visible I would just write it simpler:
```suggestion
                    Some(health) if *health == ic_management_types::Status::Healthy => None,
                    _ => Some(n) 
```